### PR TITLE
Provide access to saga completion promise

### PR DESCRIPTION
**Pull request by simlrh**
For SSR it can be useful to run some sagas and await their completion before rendering the page. Eg:

```typescript
// Run apiSaga
const sagaTask = sagaMiddleware.run(apiSaga);
// Send an action apiSaga is waiting for
store.dispatch({ type: 'GET_DATA_FROM_API' });
// Cancel all sagas still waiting for actions
store.dispatch(END);
// Await completion of sagas not waiting for actions
// This allows apiSaga to finish the api query and 'put' the data to the store
await sagaTask.done;
// Render the page including the api data
renderSSR();
```

This PR adds a function to the `redux-dynamic-modules-saga` `IExtension` to get a promise for the completion of all currently running sagas, allowing the above pattern when using `redux-dynamic-modules`. Eg:

```typescript
const sagaExtension = getSagaExtension({});
const store: IModuleStore<IState> = createStore(
  {
    initialState: {},
    enhancers: [],
    extensions: [sagaExtension],
  },
  getApiModule()
);
store.dispatch({ type: 'GET_DATA_FROM_API' });
store.dispatch(END);
await sagaExtension.done();
renderSSR();
```

### CODE REVIEW

**Comment by simlrh** [comment]
Unfortunately there was a breaking change in `redux-saga@1.0.0` where they moved `Task.done` to `Task.toPromise()`. I've added the last commit to handle this but it's a bit hacky.

File: packages/redux-dynamic-modules-saga/src/SagaManager.ts [hunk 1]
```diff
@@ -38,6 +46,23 @@ export function getSagaManager(
...
+            return tasks.keys
+                .map(k => {
+                  const task = tasks.get(k);
+                  if ('done' in task) {
+                    return task.done;
+                  } else {
+                    return task.toPromise();
+                  }
+                })
+                .reduce(
```
Comment (abettadapur) [inline packages/redux-dynamic-modules-saga/src/SagaManager.ts]: Isn't this just Promise.all?

Comment (abettadapur) [reply packages/redux-dynamic-modules-saga/src/SagaManager.ts]: Oh, you are implementing allSettled, right?

**Comment by abettadapur** [comment]
@simlrh  Just sign the CLA and I will merge!

**Comment by simlrh** [comment]
Done

**Comment by aslanyi** [comment]
Any progress?

**Comment by fostyfost** [comment]
Hey. Look at my example. It may be useful to you. And you can learn some new ideas on how to improve your library.
https://github.com/fostyfost/next-with-redux-dynamic-modules

**Comment by Tsouef** [comment]
Hey, any news ?

**Pull request closed without merging**